mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -1161,6 +1212,13 @@ LogicalResult CallOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
       return emitOpError() << "'" << calleeName.getValue()
                            << "' does not reference a valid LLVM function";
 
+    if (fn.getCConv() != getCConv()) {
+      auto fnCConv = stringifyCConv(fn.getCConv());
+      auto callCConv = stringifyCConv(getCConv());
+      return emitOpError() << "calling convention mismatch: " << callCConv
+                           << " != " << fnCConv;
+    }
```

Comment:
I don't think we should be verifying this property. The LLVM documentation only marks this as being "Undefined Behaviour" but not as an ill-formed structure. Turning undefined behaviour into a verification error is problematic as actual undefined behaviour only occurs once that instruction were to be executed. Furthermore, there could be perfectly valid code that would cause verification failure due to the above:
```mlir
%0 = isFastCC()
%1 = getCallee()
if %0 {
  call fastcc void %1()
} else {
  call void %1()
}
```
If optimizations managed to turn `getCallee` into a `SymbolRefAttr` the above program would suddenly cause verification failure despite never actually doing anything undefined.

Note that LLVM does not verify this property either: https://godbolt.org/z/W9Y4x9ezn
See also: https://mlir.llvm.org/getting_started/DeveloperGuide/#ir-verifier

Reply:
> If optimizations managed to turn getCallee into a SymbolRefAttr the above program would suddenly cause verification failure despite never actually doing anything undefined.

Ah yeah, that’s a classic problem, and it makes sense why this is UB rather than invalid; in that case, I’ll also go ahead and remove the test case I added to `invalid.mlir`.